force-app/main/default/classes/ServiceScheduleDomain_TEST.cls

```diff
@@ -43,6 +43,37 @@ private with sharing class ServiceScheduleDomain_TEST {
         );
     }
 
+    @IsTest
+    private static void shouldUpdateScheduleWithoutRelatedRecords() {
+        ServiceSchedule__c originalSchedule = [
+            SELECT Id, Name
+            FROM ServiceSchedule__c
+            LIMIT 1
+        ];
+
+        String originalName = originalSchedule.Name;
+
+        ServiceScheduleModel model = new ServiceScheduleModel(originalSchedule.Id, null);
+        model.serviceSchedule.Name = originalName + ' (Updated)';
+
+        Test.startTest();
+        domain.insertModel(model);
```

Comment:
Should this be changed to upsertModel?

Reply:
I believe so (although, with these updates mis-naming is going to run rampant throughout the entire "Service Schedule Creator" framework)